kernel/src/platform/mod.rs

```diff
@@ -111,6 +112,13 @@ pub trait Chip {
     /// chip and resumes the scheduler.
     fn sleep(&self);
 
+    /// Called at startup to reset any configuration from previous stages.
```

Comment:
So if a board doesn't call it then we are still in the same position as we are today (where it isn't called). The idea of calling it is that then we know we are starting with a clean slate.

I don't think it can go in `tock-rt0` as it needs the full PMP/MPU implementation.

Reply:
Right -- some boards need to call it because their initialization sequence has MPU state to clean out. But this suggests it shouldn't be in the chip-independent trait. The trait is for what chip-independent (e.g., kernel) code needs to do. The MPU implementations can always have additional functionality.

Reply:
I have removed the mention of calling it at boot.

Reply:
Note, that I have still left this function in the trait as I think it's useful. If people feel strongly I can remove it, but it seems worth having, then `Chips` can call it as required.